src/components/Change.vue

```diff
@@ -0,0 +1,168 @@
+<template>
+  <div class="change-modal">
+    <div class="modal-header">
+      <button
+        @click="$emit('close')"
+        type="button"
+        class="close"
+        data-dismiss="modal"
+        aria-label="Close"
+      >
+        <span aria-hidden="true">&times;</span>
+      </button>
+      <h3 class="modal-title">
+        <a v-bind:href="$attrs.change.revlink" target="_blank">{{
+          $attrs.change.comments.split('\n')[0]
+        }}</a>
+      </h3>
+    </div>
+    <div class="modal-body">
+      <h4>Author:</h4>
+      {{ $attrs.change.author }}
+      <h4>Timestamp:</h4>
+      {{ dt($attrs.change.when_timestamp) }}
+      <h4>Comments:</h4>
+      <code>
+        <div
+          class="comments"
+          v-for="(line, index) in $attrs.change.comments.split('\n')"
+          :key="index"
+        >
+          <div v-html="createLinks(line)"></div>
+        </div>
+      </code>
+      <h4>Changed files:</h4>
+      <div v-for="file in $attrs.change.files" :key="file">
+        {{ file }}
+      </div>
+      <h4>Builds:</h4>
+      <div
+        v-for="(cbs, index) in map(
+          $attrs.change.changeid,
+          $attrs.change.sourcestamp.ssid
+        )"
+        :key="index"
+      >
+        <div v-if="cbs.buildid">
+          Buildid: {{ cbs.buildid }}, Builderid: {{ cbs.builderid }}, State:
+          {{ cbs.state_string }}
+        </div>
+        <div v-else>None</div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Autolinker from 'autolinker'
+
+const baseURL = 'https://github.com/macports/macports-ports/commit/'
+
+export default {
+  name: 'Change',
+  data() {
+    return {
+      baseURL: baseURL
+    }
+  },
+  methods: {
+    getResult: function(b) {
+      let result
+      if (!b.complete && b.started_at >= 0) {
+        result = 'pending'
+      } else {
+        switch (b.results) {
+          case 0:
+            result = 'success'
+            break
+          case 1:
+            result = 'warnings'
+            break
+          case 2:
+            result = 'failure'
+            break
+          case 3:
+            result = 'skipped'
+            break
+          case 4:
+            result = 'exception'
+            break
+          case 5:
+            result = 'cancelled'
+            break
+          default:
+            result = 'unknown'
+        }
+      }
+      return result
+    },
+    map: function(changeid, ssid) {
+      var buildsets = this.$attrs.buildsets
+      var builds = this.$attrs.builds
+      var changeBuildsets = []
+      for (const buildset of buildsets) {
+        for (const sourcestamp of buildset.sourcestamps) {
+          if (sourcestamp.ssid === ssid) {
+            buildset['changeid'] = changeid
+            changeBuildsets.push(buildset)
```

Comment:
You can index changeBuildsets by bsid, and avoid the double for in next iteration

Reply:
I didn't get it. Do you mean usage of forin loop?